platforms/software/software-diagnostics/src/main/java/org/gradle/api/plugins/SoftwareReportingTasksPlugin.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.plugins;
+
+import org.apache.groovy.lang.annotation.Incubating;
+import org.gradle.api.Action;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.tasks.TaskContainer;
+import org.gradle.api.tasks.diagnostics.ArtifactTransformsReportTask;
+import org.gradle.api.tasks.diagnostics.BuildEnvironmentReportTask;
+import org.gradle.api.tasks.diagnostics.DependencyInsightReportTask;
+import org.gradle.api.tasks.diagnostics.DependencyReportTask;
+import org.gradle.api.tasks.diagnostics.OutgoingVariantsReportTask;
+import org.gradle.api.tasks.diagnostics.ResolvableConfigurationsReportTask;
+import org.gradle.api.tasks.diagnostics.TaskReportTask;
+import org.gradle.api.tasks.diagnostics.DiagnosticsTaskNames;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Adds various reporting tasks that provide information about a software project, such as information about
+ * configurations and dependency reporting tasks that make use of dependency management.
+ *
+ * @since 8.13
+ */
+@Incubating
+@Nonnull
+public abstract class SoftwareReportingTasksPlugin implements Plugin<Project>, DiagnosticsTaskNames {
```

Comment:
I'm not sure I'm a fan of implementing `DiagnosticsTaskNames`, especially with that class exposing non-software diagnostics task names

Reply:
Agreed, it's just a temp workaround.